src/shared/components/hooks/usePost.tsx

```diff
@@ -0,0 +1,49 @@
+import { useQuery } from "@tanstack/react-query";
+import axios from "axios";
+import { useEffect } from "react";
+import { useState } from "react";
+
+interface Post {
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+}
+
+const getData = async (): Promise<Post[]> => {
+  const response = await axios.get(
+    "https://jsonplaceholder.typicode.com/posts"
+  );
+  return response.data;
+};
+
+export function usePosts(isEnabled: boolean) {
+  const [search, setSearch] = useState("");
+  const [selectedUser, setSelectedUser] = useState("");
+
+  const { data, isLoading, isSuccess, isError, error } = useQuery<Post[]>({
+    queryKey: ["posts"],
```

Comment:
queryKey need to be moved to a separate file with constants

Reply:
its not error, just best practice

Reply:
feated